File: pinyougou-parent/pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service('cartService',function ($http) {
    this.findCartList=function () {
        return $http.get('/cart/findCartList.do');
    };

    this.addGoodsToCartList=function (itemId,num) {
        return $http.get('/cart/addGoodsToCartList.do?itemId='+itemId+'&num='+num);
    };

    this.findAddressList=function () {
        return $http.get('address/findListByLoginUser.do');
    };

    this.submitOrder=function (order) {
        return $http.post('order/add.do',order);
    };
    
});

File: pinyougou-parent/pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.pojogroup.Cart;

import java.util.List;

public interface CartService {

    /**
     * 添加商品到购物车中
     * @param cartList
     * @param itemId
     * @param num
     * @return
     */
    List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);

    List<Cart> findCartListFromRedis(String username);

    void addCartListToRedis(List<Cart> cartList,String username);

    List<Cart> merge(List<Cart> cookieCartList, List<Cart> redisCartList);
}
